### PR TITLE
Kaldorei Character View

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:

Proposing the addition of a custom character view for Night Elf characters to make the game experience a bit more immersive.

- **Mar 28, 2023**
  - 5afdb09c02d41525f9c6c1b9a3ca785e91899ebc
    - Designed a custom character view for **_Night Elf_** characters.
      - Added `has_kaldorei_illustration_trigger`.
      - Added `kaldorei.dds` character view art generated by Midjourney.
      - Added `environment_character_kaldorei` lighting environment.

<details><summary>Images (updated Mar 28, 2023)</summary>

image
image
image

</details>

## To-do list:

- [ ] Have a coder take a look at the code in `wc_illustration_triggers.txt` and make it more intuitive. **The current code is placeholder and not functioning correctly** (see ⚠️ **_Known issue 1_**). _(Ideally, whether or not the game chooses this background for a character shouldn't depend on their or their liege's choice of capital.)_

## ⚠️ Known issues:

- [x] 1. The background is currently not used by all Night Elves even though they are vassals of the Kaldorei Empire. (Fixed in f8e51dcc141b315f577a1467a6f41000f1bd9cda on **Apr 9 2023**)
<details><summary>Expand</summary>

image

</details>

## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:
1. To view in-game, downgrade your **Crusader Kings III** to `1.5.0.2` and look for **_Night Elf_** characters on the map.
2. Provide feedback on the lighting and background art used. Is the lighting too soft, too harsh, not coming from the right direction? _etc._
3. Report any bugs if you encounter them.

### CODE REVIEW

**Comment by ValianBlue** [comment]
<hr>

>Provide feedback on the lighting and background art used. Is the lighting too soft, too harsh, not coming from the right direction? etc.

_Overall the background art looks great. The colors are vibrant, but not too vivid and represent the Night Elf environment accurately. I assume you've searched the official art first, otherwise I would recommend looking through Blizzard's artworks to see if there is a piece that could fit._

<details>
<summary>Image</summary>

image
</details>

<hr>

>Report any bugs if you encounter them.

_I created a custom Night Elf character, a vassal under Oktone of Starfall, to see if the bug you already mentioned persisted. It did._

<details>
<summary>Image</summary>

image
</details>

_Additionally even as a top-level ruler (no liege) the background is still not present. I attempted the next custom character over a feudal kingdom to see if it was linked to government-type, but this does not seem to be the case (unless it's only linked to a **theocracy**—explaining why only Tyrande has the background as a ruler)._

<details>
<summary></summary>

image
</details>

**Comment by ercarp** [comment]
> I assume you've searched the official art first, otherwise I would recommend looking through Blizzard's artworks to see if there is a piece that could fit.

I did indeed, but I couldn't find anything official that would fit the art style of our other background arts. And the ones that were sort of okay weren't official art, so I would hesitate to use those.

However, if anyone finds a better background, I'm all for changing it. This isn't final.

File: common/scripted_triggers/wc_illustration_triggers.txt [hunk 1]
```diff
@@ -91,4 +91,18 @@ has_lordaeron_illustration_trigger = {
...
 	}
+}
+has_kaldorei_illustration_trigger = {
+	trigger_if = {
+		limit = { is_ruler = yes }
+		culture = { has_building_gfx = night_elven_building_gfx }
+	}
+	trigger_else_if = {
+		limit = { exists = host }
+		culture = { has_building_gfx = night_elven_building_gfx }
```
Comment (zumbak04) [inline common/scripted_triggers/wc_illustration_triggers.txt:102]:
```suggestion
		host.culture = { has_building_gfx = night_elven_building_gfx }
```

File: common/scripted_triggers/wc_illustration_triggers.txt [hunk 2]
```diff
@@ -91,4 +91,18 @@ has_lordaeron_illustration_trigger = {
...
+	}
+	trigger_else = {
+		exists = liege
+		culture = { has_building_gfx = night_elven_building_gfx }
```
Comment (zumbak04) [inline common/scripted_triggers/wc_illustration_triggers.txt:106]:
```suggestion
		liege.culture = { has_building_gfx = night_elven_building_gfx }
```

**Review by zumbak04 — APPROVED** [review]
Fix these and good to go.

**Pull request merged**